Google.Cloud.EntityFrameworkCore.Spanner/Storage/Internal/SpannerRetriableTransaction.cs

```diff
@@ -286,16 +288,29 @@ internal async Task RetryAsync(SpannerException abortedException, CancellationTo
         /// <returns>Returns the UTC timestamp when the data was written to the database.</returns>
         public async Task<DateTime> CommitAsync(CancellationToken cancellationToken = default)
         {
+            var tracer = TracerProviderExtension.GetTracer();
+            using var span = tracer.StartActiveSpan(TracerProviderExtension.SPAN_NAME_COMMIT);
             while (true)
             {
                 try
                 {
                     _commitTimestamp = await SpannerTransaction.CommitAsync(cancellationToken).ConfigureAwait(false);
+                    span.SetStatus(OpenTelemetry.Trace.Status.Ok);
+                    span.End();
                     return (DateTime)_commitTimestamp;
                 }
                 catch (SpannerException e) when (e.ErrorCode == ErrorCode.Aborted)
                 {
+                    span.SetAttribute(TracerProviderExtension.ATTRIBUTE_NAME_RETRYING, e.Message);
                     await RetryAsync(e, cancellationToken).ConfigureAwait(false);
+                    span.SetStatus(OpenTelemetry.Trace.Status.Ok);
```

Comment:
We should not end the span here, as that would also cause the span to be ended multiple times in case of a transaction retry.

The pseudo code of a commit call should be:

```csharp
var span = StartSpan();
while (true)
{
  try
  {
    Commit();
    span.SetStatus(OK);
    span.End();
  }
  catch (Aborted)
  {
    span.SetAttribute("Retrying");
    RetryAsync(); // This retries the transaction up until the commit call, but not including the commit call.
    // No span end here, as we are now falling back into the while(true) loop to retry the commit call.
  }
  catch (Exception e)
  {
    span.SetStatus(Exception);
    span.End();
  }
}
```

Reply:
Done. Thanks!!